File: tests/machine/x/dart/dataset_where_filter.dart
var people = [
  {'name': 'Alice', 'age': 30},
  {'name': 'Bob', 'age': 15},
  {'name': 'Charlie', 'age': 65},
  {'name': 'Diana', 'age': 45},
];

var adults = (() {
  var _q0 = <dynamic>[];
  for (var person in people) {
    if (!((person['age'] as num) >= 18)) continue;
    _q0.add({
      'name': person['name'],
      'age': person['age'],
      'is_senior': (person['age'] as num) >= 60,
    });
  }
  return _q0;
})();

void main() {
  print('--- Adults ---');
  for (var person in adults) {
    print(
      [
        person['name'],
        'is',
        person['age'],
        (person['is_senior'] ? ' (senior)' : ''),
      ].join(' '),
    );
  }
}

File: tests/machine/x/swift/dataset_where_filter.swift
struct AgeName: Equatable {
    var age: Int
    var name: String
}

var people = [AgeName(age: 30, name: "Alice"), AgeName(age: 15, name: "Bob"), AgeName(age: 65, name: "Charlie"), AgeName(age: 45, name: "Diana")]
var adults = people.compactMap { person in person.age >= 18 ? (["name": person.name, "age": person.age, "is_senior": person.age >= 60]) : nil }
print("--- Adults ---")
for person in adults as! [[String:Any]] {
    print(person["name"]!, "is", person["age"]!, person["is_senior"] as! Bool ? " (senior)" : "")
}
